backends/ebpf/ebpfDeparser.cpp

```diff
@@ -0,0 +1,376 @@
+/*
+Copyright 2022-present Orange
+Copyright 2022-present Open Networking Foundation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "ebpfDeparser.h"
+
+namespace EBPF {
+
+DeparserBodyTranslator::DeparserBodyTranslator(const EBPFDeparser *deparser) :
+        ControlBodyTranslator(deparser), deparser(deparser) {
+    setName("DeparserBodyTranslator");
+}
+
+bool DeparserBodyTranslator::preorder(const IR::MethodCallExpression* expression) {
+    auto mi = P4::MethodInstance::resolve(expression,
+                                          control->program->refMap,
+                                          control->program->typeMap);
+    auto ext = mi->to<P4::ExternMethod>();
+    if (ext != nullptr) {
+        // We skip headers emit processing which is handled by DeparserHdrEmitTranslator
+        if (ext->method->name.name == p4lib.packetOut.emit.name)
+            return false;
+    }
+
+    return ControlBodyTranslator::preorder(expression);
+}
+
+DeparserPrepareBufferTranslator::DeparserPrepareBufferTranslator(const EBPFDeparser *deparser) :
+        ControlBodyTranslator(deparser), deparser(deparser) {
+    setName("DeparserPrepareBufferTranslator");
+}
+
+bool DeparserPrepareBufferTranslator::preorder(const IR::BlockStatement* s) {
+    for (auto a : s->components) {
+        if (auto method = a->to<IR::MethodCallStatement>()) {
```

Comment:
You already check the function name below, so it may be enough to just call visit(a) for MethodCallStatements.